lib/Driver/Driver.cpp

```diff
@@ -987,6 +992,17 @@ void Driver::buildInputs(const ToolChain &TC,
         }
       }
 
+      // Apparently the convention is that the index file path is repeated.
+      // Don't try to add it in twice.
```

Comment:
It's not repeated. The convention is that the index file path specifies one of the input files; it is not itself an input file. It's questionable whether this is the *right* convention, but it *must* be separate, not "may be separate".

Reply:
What confuses me is the input ordering:
The driver-index.swift test does essentially this:
`swiftc -### -index-file -index-file-path .../driver-index.swift  .../SwiftModuleA.swift -module-name foo`
and the frontend sees:
`swift -frontend -typecheck -primary-file .../driver-index.swift .../SwiftModuleA.swift ... -module-name foo -o foo`

How does that `driver-index.swift` get to be the first input?

Reply:
Checked with @akyrtzi and found out that the key requirement is to make the index file path primary. The reordering in the status quo is not required. See `Driver::buildInputs` and `ToolChain::canCompileInputArgumentBePrimary` for my current solution. Tx.